### ISSUE TITLE
Can't create user with alternate template / tree via the "New" action button

### ISSUE THREAD

**Issue by adrianbj** [processwire/processwire-issues#1358]
### Short description of the issue

If you have a custom user template / branch parent, when you try to create a new user via the branch parent's "New" action button, when you click save after entering the title, it redirects to:
/access/users/edit/bookmarks/bookmarks/bookmarks/bookmarks/

with a 404. If you create the user via Access > Users, then it can be edited via the page tree, but it can't be created for some reason.

### Expected behavior

Users with `user-admin` permission should be able to create a new user this way, rather than needing to go via Users.

### Steps to reproduce the issue

1. Setup alternate user template / parent
2. Try to create a new user via the "new" action page in the page list
3. Save

I have been testing this as superuser and as a user with `user-admin` permission and it's the same behavior. The other somewhat related issue though is that the "new" action button still appears even for users without `user-admin` permission.

@ryancramerdesign - can I please ask that you spend some time on this alternate user template / parent feature. There are many issues reported with it (some at least a year old). I was really hesitant to implement on my latest project and I am now regretting using it because it's a mess for site editors. Sorry to be so blunt. Thank you!

**Comment by adrianbj** [comment]
In case someone else comes across this issue, I am using these hooks to make this work. Along with this, I also set the `Name format for children` setting for the parent template to `title` and I added a `title` field to the `staff` template but set it's visibility to Hidden (not shown in editor).

This creates a nice workflow where you can create a new user without the first title/name screen. It also ensures the `title` and `name` fields are always updated to match the person's first and last name if they are changed when saving in the future.

Maybe there is a cleaner option, but this works and I need to move on :)

```
$this->wire()->addHookBefore('Pages::setupNew(template=staff)', function($event) {
    $p = $event->arguments[0];
    $p->title = 'New Staff';
    $p->name = 'new-staff';
    $event->return = $p;
});
$this->addHookBefore('Pages::saveReady(template=staff)', function($event) {
    $p = $event->arguments[0];
    if($p->first_name && $p->last_name) {
        $p->title = $p->first_name . ' ' . $p->last_name;
        $p->name = $this->wire('sanitizer')->pageName($p->first_name . ' ' . $p->last_name , Sanitizer::translate);
    }
    $event->return = $p;
});
```

**Comment by cb2004** [comment]
I love this feature but find myself having to bend it to achieve what I think should be simple things. Sub users of users was fun as well.

https://github.com/processwire/processwire-issues/issues/1223

**Comment by tobaco** [comment]
I just upgraded an installation from 3.0.132 to 3.0.177 and got the same problem. Before the upgrade it worked all as expected. But now the only way to add new users is Access > Users > New and selecting the correct template and parent.
All other ways (New in pagetree or quickadd) lead to some redirects to this weird url (`/access/users/edit/bookmarks/bookmarks/bookmarks/bookmarks/`) with a 404 error.

@adrianbj 
Are your hooks and the hidden title field prevent this issue?

Thanks!

**Comment by adrianbj** [comment]
@tobaco - yes, those are fixing things for me, but keep in mind that it might be a little specific to my situation because of the `first_name` and `last_name` fields.

**Comment by tobaco** [comment]
@adrianbj 

Thanks!
I noticed the field names, but these are no problem. I got the same fields, just other names. ;-)

I just tried it and it works like a charm again. 😍

**Comment by cb2004** [comment]
I knew I had this problem a while ago:

https://github.com/processwire/processwire-issues/issues/795

**Comment by cb2004** [comment]
Ok, I just tried setting this up on a clean site and same problem, very frustrating...

**Comment by adrianbj** [comment]
@cb2004 - did you try my hack? It's been working really well for me until @ryancramerdesign can get this fixed.

**Comment by cb2004** [comment]
@adrianbj the main problem I had was I don't need any extra fields so just pass, email, roles, admin_theme is fine for me, so really basic as we then integrate with a CRM and its data using the email address. What I had to do was set the 'Name format for children' to 'title' (even though the child template doesn't have a title), then I add a role automatically:

```
$wire->addHookAfter('Users::added(template=101)', function($e) {
	$p = $e->arguments[0];
	$p->addRole(1281);
	$p->save();
});
```
This seems to get me to where I need to be, and add a role in the process so I can hide that field.

This feature has great potential but as said above I always seem to have to hack it. I also would like users underneath users but have to hack away again for that.

Also the instructions need updating a bit now that @ryancramerdesign has changed the layout of the tabs:

https://processwire.com/blog/posts/processwire-core-updates-2.5.14/#multiple-templates-or-parents-for-users

**Comment by ryancramerdesign** [comment]
@adrianbj I was able to duplicate the behavior you mentioned and have attempted a fix for this issue. Though I don't yet understand why the two hooks you mentioned seemed to resolve the issue in your case because those hooks appear to be unrelated to the bit of code I tracked down needing a fix in ProcessPageAdd (see commit). So it's possible that there's more to it I don't understand yet, or that your setup is quite a bit different than mine. Please let me know what you find. Thanks.

**Comment by adrianbj** [comment]
I can't seem to test this properly at the moment because of the issue I just mentioned over here: https://github.com/processwire/processwire-issues/issues/1308

**Comment by ryancramerdesign** [comment]
@adrianbj This one should be fixed also with the fix posted for #1308, though let me know if you find it isn't. Thanks.

**Comment by adrianbj** [comment]
Thanks @ryancramerdesign - I think it should be fine now. 

As and FYI, my setup is not quite standard because I hide the title field and use that hook to create the title (and name) in the background after the `first_name` and `last_name` fields have been entered. I think without this it feels a bit clunky - having to enter fullname in the title and then separate first and last names. That is why I used those hooks the way I did, but it was also taking care of the error that was reported in the OP.
